src/main.cpp

```diff
@@ -93,6 +107,45 @@ static void _set_prod_option(bool& is_prod) {
 #endif
 }
 
+static void _set_tag_options(std::string& tags) {
+	if (!boost::filesystem::exists(os_release_filename) &&
+		!boost::filesystem::exists(installed_versions_filename)) {
+		cerr << "Default tag missing. Please use --tags to register the device." << endl;
+		throw std::invalid_argument("Not possible to source tags from filesystem.");
+	}
+
+	// Try to retrieve default value from filesystem
+	ptree os_tree;
+	if (boost::filesystem::exists(installed_versions_filename)) {
+		try {
+			read_json(installed_versions_filename, os_tree);
+			std::string root_node = os_tree.begin()->first;
+			for (ptree::value_type &tag : os_tree.get_child(root_node + ".custom.tags")) {
+				// "tags" is a list. Use first item as DG doesn't support a list
+				tags = tag.second.data();
+				break;
+			}
+		}
+		catch (boost::property_tree::json_parser::json_parser_error const&) {
+			throw po::error("Unable to read tags from installed_versions");
```

Comment:
Maybe fallback to reading from `os_release`?

Reply:
fixed